src/libraries/Microsoft.PowerFx.Core/Public/Types/FormulaType.cs

```diff
@@ -57,9 +58,47 @@ internal FormulaType(DType type)
             _type = type;
         }
 
+        // Entites may be recursive and their Dytype is tagged with additional schema metadata. 
+        // Expand that metadata into a proper Dtype. 
+        private static DType GetExpandedEntityType(DType expandEntityType, string relatedEntityPath)
```

Comment:
Can relationship support be a separate PR?

Reply:
ok.